rust/src/lib/ifaces/ethtool.rs

```diff
@@ -0,0 +1,160 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Default)]
+#[non_exhaustive]
+pub struct EthtoolConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pause: Option<EthtoolPauseConfig>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub feature: Option<EthtoolFeatureConfig>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub coalesce: Option<EthtoolCoalesceConfig>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ring: Option<EthtoolRingConfig>,
+}
+
+impl EthtoolConfig {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Default)]
+#[non_exhaustive]
+pub struct EthtoolPauseConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rx: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tx: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub autoneg: Option<bool>,
+}
+
+impl EthtoolPauseConfig {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Default)]
+#[serde(rename_all = "kebab-case")]
+#[non_exhaustive]
+pub struct EthtoolFeatureConfig {
+    #[serde(skip_serializing_if = "Option::is_none", alias = "rx")]
+    pub rx_checksum: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none", alias = "gro")]
+    pub rx_gro: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none", alias = "lro")]
+    pub rx_lro: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none", alias = "rxvlan")]
+    pub rx_vlan_hw_parse: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none", alias = "txvlan")]
+    pub tx_vlan_hw_insert: Option<bool>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        alias = "ntuple",
+        alias = "ntuple-filters"
+    )]
+    pub rx_ntuple_filter: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none", alias = "rxhash")]
+    pub rx_hashing: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tx_scatter_gather: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tx_tcp_segmentation: Option<bool>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        alias = "generic-segmentation-offload"
+    )]
+    pub tx_generic_segmentation: Option<bool>,
+}
+
+impl EthtoolFeatureConfig {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Default)]
```

Comment:
This is struct only bool and u32, you may enable `Copy` trait which could make the life easier as you don't need to worry about ownership of it.

Reply:
Done, thank you!